### PR TITLE
Follow-up to Explain Impersonation (PR#2020)

### PR DESCRIPTION
**Pull request by ahardin-rh**
Follow-up to https://github.com/openshift/openshift-docs/pull/2020

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@tnguyen-rh @bfallonf @tpoitras PTAL. Thanks!

File: architecture/additional_concepts/authentication.adoc [hunk 1]
```diff
@@ -88,22 +88,29 @@ the `system:anonymous` virtual user and the `system:unauthenticated` virtual
...
+which indicates that the requester wants to have the request handled as though
+it came from the specified user. This can be done on the command line by passing
+the `*--as=username*` flag.
+
+Before User A is allowed to impersonate User B, User A is first authenticated.
+Then, an authorization check occurs to ensure that User A is allowed to
+impersonate the user named User B. If User A is requesting to impersonate a
+service account (*system:serviceaccount:namespace:name*), we check to ensure
+that User A can impersonate the *serviceaccount* named *name* in *namespace*. If
+User A is not allowed, a 403 is returned.
```
Comment (tnguyen-rh) [inline architecture/additional_concepts/authentication.adoc]: - s/we check/{product-title} checks/
- Probably the last two sentences can be coalesced, something like: "... checks to ensure ... in namespace, and returns a 403 if that check fails."  Tangentially, i'm not sure if naked "a 403" is ok.  Maybe it's better to stick with the two sentences and say "If the check fails, the request fails with a 403 (Forbidden) error code." (presuming [this](https://en.wikipedia.org/wiki/HTTP_403) is the 403 in question).

File: architecture/additional_concepts/authentication.adoc [hunk 2]
```diff
@@ -88,22 +88,29 @@ the `system:anonymous` virtual user and the `system:unauthenticated` virtual
...
+
+By default, project administrators and editors are allowed to impersonate
+service accounts in their namespace. There is also a new group called
+*sudoers*. That role allows a user to impersonate *system:admin*, which in turn
```
Comment (tnguyen-rh) [inline architecture/additional_concepts/authentication.adoc]: Hmm, what does "new" mean, specifically (which version)?  Also, is it a *group* or a *role* (as stated in the next sentence)?  Can those two sentences be combined?

File: architecture/additional_concepts/authentication.adoc [hunk 3]
```diff
@@ -88,22 +88,29 @@ the `system:anonymous` virtual user and the `system:unauthenticated` virtual
...
+has cluster administrator permissions. This grants some protection against typos
+(but not security) for someone administering the cluster. For example, `oc
+delete nodes --all` would be forbidden, but `oc delete nodes --all
+--as=system:admin` would be allowed. You still have the power, but you have to
+mean it. You can add a user to that group using `oadm policy
+add-cluster-role-to-user sudoer <username>`.
```
Comment (tnguyen-rh) [inline architecture/additional_concepts/authentication.adoc]: Not sure "you have to mean it" is appropriate.  Suggest dropping that sentence entirely.

**Comment by tnguyen-rh** [comment]
@ahardin-rh Some minor touch-ups, otherwise LGTM. :+1:

**Comment by ahardin-rh** [comment]
@tnguyen-rh you rock! Thank you for your thoughtful comments.  :guitar:  :metal:

**Comment by ahardin-rh** [comment]
[rev_history]
|link:../architecture/additional_concepts/authentication.html[Authentication]
|Added the link:../architecture/additional_concepts/authentication.html#authentication-impersonation[Impersonation] section.
%

**Comment by tnguyen-rh** [comment]
@ahardin-rh FYI, i modified the revhist note to add "Additional Concepts -> ".

**Comment by ahardin-rh** [comment]
@tnguyen-rh Perfect! Thank you!

**Comment by adellape** [comment]
See https://github.com/openshift/openshift-docs/pull/2020#issuecomment-224398156

**Pull request merged**